File: Benchmarking/stats/euler15/euler15.cpp
//
//  main.cpp
//  Euler15
//
//  Created by Sonni on 20/05/16.
//  Copyright © 2016 Sonni Jensen. All rights reserved.
//

#include <iostream>

unsigned long calc(unsigned int siz)
{
    unsigned long mat[siz + 1][siz + 1];
    for (unsigned int i = 0; i < siz+1; i++)
        for (unsigned int j = 0; j < siz+1; j++)
            mat[i][j] = 1;
    
    for (int i = siz - 1; i >= 0; i--)
        for (int j = siz - 1; j >= 0; j--)
            mat[i][j] = mat[i+1][j] + mat[i][j+1];
        
    
    return mat[0][0];
}

int main(int argc, const char * argv[]) {
    std::cout << calc(atoi(argv[1])) << std::endl;;
    return 0;
}

File: Benchmarking/noscaling/euler15/euler15.py
import sys

siz = int(sys.argv[1])

def calc(siz):
  mat = [[1 for i in xrange(siz+2)] for j in xrange(siz+2)]

  for j in xrange(siz-1, -1, -1):
    for i in xrange(siz-1, -1, -1):
      mat[i][j] = mat[i][j+1] + mat[i+1][j]
  return mat[0][0]

calc(siz)
